crates/http-api-bindings/src/completion/openai.rs

```diff
@@ -40,47 +41,97 @@ impl OpenAICompletionEngine {
     }
 }
 
-#[derive(Serialize)]
+#[derive(Debug, Default, Serialize)]
 struct CompletionRequest {
     model: String,
-    prompt: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
prompt and messages fields are not allowed at the same time even with the empty value.